sailthru/sailthru_http.py

```diff
@@ -11,36 +11,33 @@ def flatten_nested_hash(hash_table):
     """
     def flatten(hash_table, brackets=True):
         f = {}
-        for key,value in hash_table.items():
-            _key = '[' + str(key) + ']' if (brackets == True) else str(key)
-            if type(value) is dict:
-                for k,v in flatten(value).items():
+        for key, value in hash_table.items():
+            _key = '[' + str(key) + ']' if brackets else str(key)
+            if isinstance(value, dict):
+                for k, v in flatten(value).items():
                     f[_key + k] = v
-            elif type(value) is list:
+            elif isinstance(value, list):
                 temp_hash = {}
-                for i,v in enumerate(value):
+                for i, v in enumerate(value):
                     temp_hash[str(i)] = v
-                for k,v in flatten(temp_hash).items():
-                    f[ _key + k] = v
+                for k, v in flatten(temp_hash).items():
+                    f[_key + k] = v
             else:
                 f[_key] = value
         return f
     return flatten(hash_table, False)
 
-def sailthru_http_request(url, data, method, file_data = None):
+def sailthru_http_request(url, data, method, file_data=None):
     """
     Perform an HTTP GET / POST / DELETE request
     """
-    files = {}
     data = flatten_nested_hash(data)
     method = method.upper()
     params = data if method != 'POST' else None
-    body = data if method == 'POST' else None
+
     try:
-        headers = { 'User-Agent': 'Sailthru API Python Client %s; Python Version: %s' % ('2.2.0', platform.python_version())}
-        response = requests.request(method, url, params = params, data = data, files = file_data, headers = headers, timeout = 10)
-        if response.status_code is None:
```

Comment:
I removed this section since response.status_code will never be None. The requests library will throw an exception if there isn't a proper HTTP response.